### PR TITLE
Eng 24222 amazon java client modification requests

### PR DESCRIPTION
**Pull request by ryanmckaytx**
https://shippo.atlassian.net/browse/ENG-24222
Add `test` attribute to all models, except:
* Manifest - response body does not have `test` attribute
* Pickup - response body does not have `test` attribute
* CarrierAccount - already had `test` attribute, but added `isTest` method to match standard access pattern

Add comments to PickupTest for why test is ignored

Update gson dependency to remove CVE

### CODE REVIEW

File: src/main/java/com/shippo/model/Address.java [hunk 1]
```diff
@@ -35,8 +35,17 @@ public class Address extends APIResource {
     Object metadata;
     Object messages;
 	ValidationResults validation_results;
+	private boolean test;
 
-    public class ValidationResults {
+    public boolean isTest() {
+		return test;
+	}
```
Comment (lambertkuang-shippo) [inline src/main/java/com/shippo/model/Address.java:42]: nit, but looks like formatting is off

File: src/main/java/com/shippo/model/Batch.java [hunk 2]
```diff
@@ -55,6 +55,8 @@ public static enum BatchStatus {
 	@SerializedName("label_filetype")
 	private LabelFileType labelFileType;
 
+	private boolean test;
```
Comment (lambertkuang-shippo) [inline src/main/java/com/shippo/model/Batch.java:58]: I didn't see it in the api ref, but looks like it is returned in the response

File: src/main/java/com/shippo/model/Pickup.java [hunk 3]
```diff
@@ -30,6 +30,7 @@ public class Pickup extends APIResource {
 	Object timezone;
 	Object messages;
 	Object metadata;
+	// TODO: add private boolean test attribute.  Today response payload does not include test attribute.
```
Comment (lambertkuang-shippo) [inline src/main/java/com/shippo/model/Pickup.java:33]: are the docs inaccurate? It says an `is_test` attribute exists https://goshippo.com/docs/reference#pickups

**Review by lambertkuang-shippo — COMMENTED** [review]
Noticed a few inconsistencies with the docs, but we can go over them later. Also, some of the formatting looks off to me but not sure if that's just in this review. Otherwise, LGTM 👍

File: src/main/java/com/shippo/model/Address.java (hunk 1 above)
Comment (ryanmckaytx) [reply src/main/java/com/shippo/model/Address.java:42]: good call, applied standard formatting to that file

File: src/main/java/com/shippo/model/Batch.java (hunk 2 above)
Comment (ryanmckaytx) [reply src/main/java/com/shippo/model/Batch.java:58]: will circle back with an audit of the api ref

File: src/main/java/com/shippo/model/Pickup.java (hunk 3 above)
Comment (ryanmckaytx) [reply src/main/java/com/shippo/model/Pickup.java]: mapped `is_test` in payload to `test` in model

File: src/main/java/com/shippo/model/CustomsDeclaration.java [hunk 4]
```diff
@@ -38,13 +38,22 @@ public class CustomsDeclaration extends APIResource {
...
+	private boolean test;
 
     public static CustomsDeclaration create(Map<String, Object> params) throws AuthenticationException,
             InvalidRequestException, APIConnectionException, APIException {
         return create(params, null);
     }
 
-    public String getInstanceURL() {
+    public boolean isTest() {
+		return test;
```
Comment (lambertkuang-shippo) [inline src/main/java/com/shippo/model/CustomsDeclaration.java:49]: can you also apply the formatting to this file?

File: src/main/java/com/shippo/model/Parcel.java [hunk 5]
```diff
@@ -36,7 +37,15 @@ public static Parcel createForShipment(double length, double width, double heigh
         return p;
     }
 
-    public static Parcel create(Map<String, Object> params) throws AuthenticationException, InvalidRequestException,
+    public boolean isTest() {
+		return test;
```
Comment (lambertkuang-shippo) [inline src/main/java/com/shippo/model/Parcel.java:41]: same here, formatting looks off

File: src/test/java/com/shippo/model/CustomsDeclarationTest.java [hunk 6]
```diff
@@ -64,6 +64,7 @@ public void testValidCreate() {
 		Shippo.setDEBUG(true);
         CustomsDeclaration testObject = createCustomsDeclarationFixture();
         assertEquals(testObject.getObjectState(), "VALID");
+		assertEquals(Shippo.apiKeyIsTest, testObject.isTest());
```
Comment (lambertkuang-shippo) [inline src/test/java/com/shippo/model/CustomsDeclarationTest.java:67]: formatting

File: src/test/java/com/shippo/model/CustomsItemTest.java [hunk 7]
```diff
@@ -35,6 +36,7 @@ public class CustomsItemTest extends ShippoTest {
     @Test
     public void testValidCreate() {
         CustomsItem testObject = createCustomsItemFixture();
+		assertEquals(Shippo.apiKeyIsTest, testObject.isTest());
```
Comment (lambertkuang-shippo) [inline src/test/java/com/shippo/model/CustomsItemTest.java:39]: .

File: src/test/java/com/shippo/model/ShippoTest.java [hunk 8]
```diff
@@ -27,4 +31,15 @@ public Map<String, Object> getInvalidObjectMap() {
     public static Object nullToEmptyString(Object value) {
         return value == null ? "" : value;
     }
+
+	// Make sure we get a test rate.  We are using a test auth token, so it should not be possible to get a non-test rate back,
+	// but previous author was uncertain, and I don't know enough about the underlying implementation to be sure.
+	static Rate selectTestRate(List<Rate> rateList) {
```
Comment (lambertkuang-shippo) [inline src/test/java/com/shippo/model/ShippoTest.java:37]: .

File: src/test/java/com/shippo/model/TransactionTest.java [hunk 9]
```diff
@@ -22,6 +23,7 @@ public class TransactionTest extends ShippoTest {
 	public void testValidCreate() {
 		Transaction testObject = createTransactionFixture();
 		assertEquals("SUCCESS", testObject.getStatus());
+        assertEquals(Shippo.apiKeyIsTest, testObject.isTest());
```
Comment (lambertkuang-shippo) [inline src/test/java/com/shippo/model/TransactionTest.java:26]: .

**Review by lambertkuang-shippo — COMMENTED** [review]
just a few more formatting inconsistencies, but lgtm otherwise!

**Review by lambertkuang-shippo — APPROVED** [review]
LGTM, thanks for formatting it

**Pull request merged**